### ISSUE TITLE
getCameraMatrix does not return up-to-date FoV value

### ISSUE THREAD

**Issue by quiret** [multitheftauto/mtasa-blue#1282]
**Describe the bug**
When calling `local sx, sy, sz, tx, ty, tz, roll, fov = getCameraMatrix()` and inspecting the fov value, then this value is not up-to-date.

Symptoms:
1) setCameraFieldOfView does not synchronize with getCameraMatrix FoV
2) accelerating in a vehicle increases the FoV, but not the value returned by getCameraMatrix
3) camera calculation code is wrong if FoV changes from default value

**To reproduce**
1) the scripting way (if not in vehicle):
```
setCameraFieldOfView("player", 90)

setTimer(function()
    local _,_,_,_,_,_,_,fov = getCameraMatrix()
    assert(fov == 90)
end, 2000, 1)
```

2) the gameplay way: enter a vehicle, accelerate very fast, then execute the following command handler (clientside):
```
addCommandHandler("getfov", function()
    local _,_,_,_,_,_,_,fov = getCameraMatrix()
    outputChatBox("current field of view is " .. fov);
end)
```

**Expected behaviour**
The field of view is supposed to adjust to the camera's real value instead of being default. If FoV is changed by script to 90 getCameraMatrix should also return the interpolating value.

**Version**
MTA:SA 1.5.7

**Comment by qaisjp** [comment]
Good point. I'd say [[set](https://wiki.multitheftauto.com/wiki/SetCameraFieldOfView)|[get](https://wiki.multitheftauto.com/wiki/GetCameraFieldOfView)]CameraFieldOfView should return the static setting, but [getCameraMatrix](https://wiki.multitheftauto.com/wiki/SetCameraMatrix) the actual 'realtime' FOV.

Quick research shows that, client side, the `fov` param does this:

```cpp
fFOV = m_pCamera->GetFOV();
```

And `CClientCamera::GetFOV()` is this:

```cpp
float GetFOV() { return m_fFOV; }
```

Doesn't seem like we use `CCam::GetFOV()` anywhere (but we do use `CCam::SetFOV()`, so I guess this might be enough:

```cpp
CCam* pCam = m_pCamera->GetCam(m_pCamera->GetActiveCam());
pCam->GetFOV();
```

Server side only works for fixed cameras, so not relevant.

**Comment by quiret** [comment]
Good idea, qaisjp! If you can provide me with a nightly build with the fix inserted then I am willing to test it.

**Comment by qaisjp** [comment]
You can get a build once CI has passed on #1283. You need to click on the "Details" button next to one of the Checks and then download a build via the "Artifacts" menu. It looks like this:

![image](https://user-images.githubusercontent.com/923242/76164942-dce9b780-614a-11ea-9409-4ada4d6a6cdb.png)


I've given it a quick test and it works fine both on foot, in vehicle and in fixed camera mode.

I think the only controversial thing about this change would be slightly changing the behaviour of FOV. But, in my opinion, I think it's OK because I think (semantically) people expect setCameraMatrix to be a representation of the current camera state. And people can still use the individual functions to get the individual settings.

**Comment by quiret** [comment]
I cannot test this build because I have no compatible netc.dll module:
![network_module_error](https://user-images.githubusercontent.com/7682485/76165295-1a077700-6156-11ea-81c9-5c80c7a4e76e.png)
Could you please provide me with something that works? I see that the build has not been uploaded to nightly.mtasa.com yet.

**Comment by qaisjp** [comment]
Nightly builds only run for master. So we have to wait until my PR is merged (+ 1 day for nightly to build).

I don't know why the CI build doesn't work. Did you download `InstallFiles.zip`, extract it to its own folder, and then run `MTA San Andreas.exe`?

**Comment by qaisjp** [comment]
Nevermind i can't get the CI build to work either

**Comment by quiret** [comment]
I searched for netc.dll in this repository and found working uploaded files at:
https://mirror.mtasa.com/bdata/net.dll
https://mirror.mtasa.com/bdata/netc.dll

Thank you very much for your effort, qaisjp! 😁👍 I hope that we can get this patch into the official MTA:SA release ASAP to improve the product. The fix does work as expected. There are still other things that need addressing but those will be put into separate issues.

**Comment by qaisjp** [comment]
Yeah it's weird, I thought our pipeline downloads those files and adds it to the archive. Oh well. Glad it worked.

Please do send over those issues. And pull requests.

**Comment by quiret** [comment]
> Please do send over those issues. And pull requests.

Dear @qaisjp,
I want to inform you that I still plan to work on the MTA codebase, but there are pressing issues with GTA:SA itself that need addressing with good scientific research. At one point I did start to believe that remaking the engine, in parts, is required. I am very happy that you prioritized my issue after I started helping people in the scripting section. That research will still take months so I continue to be sorry that this is as much as I can and am willing to do. I will hang around as helpful entity and commentator tho.
Kind regards, Martin

**Comment by Lpsd** [comment]
@quiret https://github.com/saml1er/gta-reversed may be of interest to you

**Comment by FileEX** [comment]
Fixed in ac7351f
